### ISSUE TITLE
1.5.0 does not show emoji icons anymore

### ISSUE THREAD

**Issue by batrand** [FlorianWoelki/obsidian-icon-folder#140]
**Describe the bug**
I updated to a latest version and all the icons I set to my files and folders no longer display. They are all set using emojis. I had to restore the plugin folder from a different device to see them again. It happens for an empty vault, too

**To Reproduce**
1. Create new vault and install plugin
![image](https://user-images.githubusercontent.com/8514709/209087803-465f9202-3f75-4e20-befe-d96e9c6b317a.png)

2. Right click on a note, change icon, select any that shows or enter an emoji
![image](https://user-images.githubusercontent.com/8514709/209087849-870c9481-82aa-4c4e-98e7-329cb28dcf73.png)

**Expected behavior**
The icon should show in the file tree, but the file tree does not show anything
![image](https://user-images.githubusercontent.com/8514709/209088017-e20c785a-77b7-4cf5-933b-0287c8020972.png)

**Screenshots**

**Additional context**
Not sure if relevant, but I compared the data.json in the new empty vault, and notice that it does not add any icon there anymore:
![image](https://user-images.githubusercontent.com/8514709/209088173-302bddca-2a98-4f06-b490-be9ca384b95b.png)

Compared to the one in my working vault where I restored the plugin folder from another device:
![image](https://user-images.githubusercontent.com/8514709/209088390-a054e210-e575-480b-9980-4104a0a1800b.png)

**Comment by Ellpeck** [comment]
Same here, kind of? 

In my `data.json` file, all of the icon mappings are still there as they previously were, but none of them display in my vault, and I get this exception on startup:

```
TypeError: Cannot read properties of null (reading 'hasAttribute')
    at colorizeIcon (plugin:obsidian-icon-folder:2243:21)
    at customizeIconStyle (plugin:obsidian-icon-folder:2226:12)
    at insertIconToNode (plugin:obsidian-icon-folder:2604:26)
    at eval (plugin:obsidian-icon-folder:2287:25)
    at Array.forEach (<anonymous>)
    at eval (plugin:obsidian-icon-folder:2272:14)
    at Array.forEach (<anonymous>)
    at addIconsToDOM (plugin:obsidian-icon-folder:2262:19)
    at IconFolderPlugin.handleChangeLayout (plugin:obsidian-icon-folder:3787:9)
    at eval (plugin:obsidian-icon-folder:3695:82)
    at t.e.tryTrigger (app.js:1:1045441)
    at t.e.trigger (app.js:1:1045374)
    at t.trigger (app.js:1:1812523)
    at t.updateLayout (app.js:1:1805262)
    at r (app.js:1:1756801)
```

**Comment by elvarb** [comment]
Obsidian crashed for me just now after I renamed a folder.
Restarted Obsidian and now only a handful of my folders have the icons set. Seems that only folders I created long time ago and defined an icon for them still have the icons, the newer ones do not. Second time this has happened.

No exception shows up for me.

In the data.json file It seems that the folders defined before the one I renamed are showing up ok. 
The folder I renamed is still listed there using the old name. The new name though worth noting included a comma.

I renamed the folder back to the original name.

Now when going through the data.json file again and comparing it to the folder icons set after restarting Obsidian it seems to stop with a folder with an Icelandic character in its name.

I'm guessing there is some JSON parsing problems probably related to encoding. Same could possibly explain the problem with emoji's.
